element_deeplabcut/dlc.py

```diff
@@ -301,26 +255,22 @@ class BodyPart(dj.Lookup):
     """
 
     @classmethod
-    def insert_all_from_model(cls, key, body_part_list=None,
-                              description_list=None,
-                              skip_duplicates=True):
-        """ Insert all body parts from a given model
-        To see body_part list before inserting, use
-        `element_deeplabcut.readers.dlc_reader.PoseEstimation(config_path).body_parts`
-        :param key: specifying one model
-        :param description_list: optional list of string descriptions of each part
-        :parm skip_duplicates: skip if already in table
-        """
-        if not body_part_list:
-            from .readers.dlc_reader import PoseEstimation
-            project_path = find_full_path(get_dlc_root_data_dir(),
-                                          get_session_directory(key))
-            body_parts = PoseEstimation(project_path).body_parts
-        if not description_list:
-            description_list = [''] * len(body_parts)
-        for i in range(len(body_parts)):
-            cls.insert1((body_parts[i], description_list[i]),
-                        skip_duplicates=skip_duplicates)
+    def insert_from_config(cls, dlc_config: dict):
+        # handle dlc_config being a yaml file
+        if not isinstance(dlc_config, dict):
+            dlc_config_fp = pathlib.Path(dlc_config)
+            if dlc_config_fp.exists() and dlc_config_fp.suffix in ('.yml', '.yaml'):
+                with open(dlc_config, 'rb') as f:
+                    dlc_config = yaml.safe_load(f)
+        # -- Check and insert new BodyPart --
+        if 'bodyparts' in dlc_config:
+            tracked_body_parts = cls.fetch('body_part')
+            new_body_parts = np.setdiff1d(dlc_config['bodyparts'], tracked_body_parts)
+            if new_body_parts:
+                print(f'Existing body parts: {tracked_body_parts}')
+                print(f'New body parts: {new_body_parts}')
+                if dj.utils.user_choice(f'Insert {len(new_body_parts)} new body part(s)?') == 'yes':

```

Comment:
@kabilar at once point provided the feedback that I should avoid user choice in element methods. I think this case differs in that it's instead asking for confirmation, not a choice.